### PR TITLE
[Inference]add pir log in running pir pass

### PR DESCRIPTION
**Pull request by bukejiyu**
### PR types

Others
### PR changes

Others
### Description

移步->https://github.com/PaddlePaddle/Paddle/pull/59825
add pir log
image
### Others
Pcard-71500

### CODE REVIEW

File: paddle/pir/pass/pass.h [hunk 1]
```diff
@@ -90,6 +91,14 @@ class IR_API Pass {
 
   virtual bool Initialize(IrContext* context) { return true; }
 
+  void PrintPassMatchCount(int64_t pass_match_count) {
```
Comment (yuanlehome) [inline paddle/pir/pass/pass.h]: PrintStatistics

File: paddle/pir/pass/pass.h [hunk 2]
```diff
@@ -19,6 +19,7 @@
 #include <vector>
 
 #include "paddle/common/enforce.h"
+#include "paddle/fluid/string/pretty_log.h"
```
Comment (yuanlehome) [inline paddle/pir/pass/pass.h]: 使用这个头文件paddle/utils/string/pretty_log.h

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc [hunk 3]
```diff
@@ -54,8 +54,9 @@ class GreedyPatternRewriteDriver : public pir::PatternRewriter {
     }
   }
 
-  bool Simplify() {
-    bool changed = false;
+  std::pair<bool, int64_t> Simplify() {
+    int64_t sum_pattern_match_count = 0;
```
Comment (yuanlehome) [inline paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc:58]: sum_num_rewrites

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc [hunk 4]
```diff
@@ -54,8 +54,9 @@ class GreedyPatternRewriteDriver : public pir::PatternRewriter {
...
+    int64_t cur_iterations_count = 0;
```
Comment (yuanlehome) [inline paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc]: int64_t sum_num_rewrites = 0;
int64_t num_rewrites = 0;

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc [hunk 5]
```diff
@@ -80,19 +81,18 @@ class GreedyPatternRewriteDriver : public pir::PatternRewriter {
...
       }
 
-      changed = ProcessWorklist();
-    } while (changed);
-
-    return !changed;
+      cur_iterations_count = ProcessWorklist();
+      sum_pattern_match_count += cur_iterations_count;
+    } while (cur_iterations_count != 0);
+    bool converged = cur_iterations_count == 0 ? true : false;
```
Comment (yuanlehome) [inline paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc]: bool converged = cur_iterations_count == 0;

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc [hunk 6]
```diff
@@ -214,18 +212,20 @@ class GreedyPatternRewriteDriver : public pir::PatternRewriter {
...
-                           GreedyRewriteConfig config) {
+std::pair<bool, int64_t> ApplyPatternsGreedily(
+    Region& region,  // NOLINT
+    const FrozenRewritePatternSet& patterns,
+    GreedyRewriteConfig config) {
   if (!config.region) config.region = &region;
 
   GreedyPatternRewriteDriver driver(region.ir_context(), patterns, config);
-  bool converged = driver.Simplify();
+  auto converged_total_match_nums_pair = driver.Simplify();
```
Comment (yuanlehome) [inline paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc]: auto [converged, num_rewrites] = driver.Simplify();

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc [hunk 7]
```diff
@@ -214,18 +212,20 @@ class GreedyPatternRewriteDriver : public pir::PatternRewriter {
...
+  auto converged = converged_total_match_nums_pair.first;
   if (!converged) {
     LOG(WARNING) << "The pattern rewrite did not converge after scaning "
                  << config.max_iterations << " times";
   }
-  return converged;
+  return converged_total_match_nums_pair;
```
Comment (yuanlehome) [inline paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc]: return std::make_pair(converged, num_rewrites);

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.h [hunk 8]
```diff
@@ -64,23 +64,31 @@ class IR_API GreedyRewriteConfig {
...
-    failed |= !ApplyPatternsGreedily(region, patterns, config);
+    auto converged_total_match_nums_pair =
+        ApplyPatternsGreedily(region, patterns, config);
+    auto converged = converged_total_match_nums_pair.first;
+    auto pass_region_match_count = converged_total_match_nums_pair.second;
+    failed |= !converged;
+    regions_pass_match_count += pass_region_match_count;
   }
-  return !failed;
+  return std::make_pair(failed, regions_pass_match_count);
```
Comment (yuanlehome) [inline paddle/pir/pattern_rewrite/pattern_rewrite_driver.h]: 这一块和上面的逻辑一样改掉变量命名

**Review by yuanlehome — COMMENTED** [review]
存量pass均加上log print

File: paddle/pir/pass/pass.h (hunk 1 above)
Comment (bukejiyu) [reply paddle/pir/pass/pass.h]: 已改

File: paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc (hunk 4 above)
Comment (bukejiyu) [reply paddle/pir/pattern_rewrite/pattern_rewrite_driver.cc]: 已改

File: paddle/fluid/pir/transforms/inplace_pass.cc [hunk 9]
```diff
@@ -469,14 +469,19 @@ class InplacePass : public pir::Pass {
       kv.first->set_attribute(
           "is_inplace",
           pir::BoolAttribute::get(pir::IrContext::Instance(), true));
+      num_rewrites_++;
     }
+    PrintStatistics(num_rewrites_);
     LOG_FIRST_N(INFO, 1)
         << "Apply inplace pass on lowering ::pir::Program to Kernel Dialect.";
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/inplace_pass.cc]: 删掉

File: paddle/fluid/pir/transforms/sub_graph_extract_pass.cc [hunk 10]
```diff
@@ -49,12 +49,13 @@ class SubGraphExtractPass : public pir::Pass {
...
-    IR_ENFORCE(module_op, "build_cinn_pass should run on module op.");
+    IR_ENFORCE(module_op, "sub_graph_extract_pass should run on module op.");
     auto& block = module_op.block();
 
     std::vector<GroupOpsVec> groups =
         ::pir::SubgraphDetector(&block, IsSplitOp)();
-    LOG(INFO) << "--- [build_cinn_pass] detected " << groups.size()
+    PrintStatistics(static_cast<int64_t>(groups.size()));
+    LOG(INFO) << "--- [sub_graph_extract_pass] detected " << groups.size()
               << " cinn supported subgraphs";
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/sub_graph_extract_pass.cc]: 删掉

File: paddle/fluid/pir/transforms/build_cinn_pass.cc [hunk 11]
```diff
@@ -195,6 +195,7 @@ class BuildCinnPass : public pir::Pass {
 
     std::vector<GroupOpsVec> groups =
         ::pir::SubgraphDetector(&block, IsSupportCinn)();
+    PrintStatistics(static_cast<int64_t>(groups.size()));
     LOG(INFO) << "--- [build_cinn_pass] detected " << groups.size()
               << " cinn supported subgraphs";
```
Comment (yuanlehome) [inline paddle/fluid/pir/transforms/build_cinn_pass.cc]: 删掉

File: paddle/pir/pass/pass.h [hunk 12]
```diff
@@ -90,6 +91,19 @@ class IR_API Pass {
 
   virtual bool Initialize(IrContext* context) { return true; }
 
+  void PrintStatistics(int64_t pass_match_count) {
+    if (pass_match_count > 0) {
```
Comment (yuanlehome) [inline paddle/pir/pass/pass.h]: 等于0应该也需要打印出来

**Review by yuanlehome — DISMISSED** [review]
LGTM

File: paddle/pir/pass/pass.cc [hunk 13]
```diff
@@ -97,11 +97,12 @@ bool detail::PassAdaptor::RunPass(Pass* pass,
   pass->pass_state_ = PassExecutionState(op, am);
 
   PassInstrumentor* instrumentor = am.GetPassInstrumentor();
-
   if (auto* adaptor = dynamic_cast<PassAdaptor*>(pass)) {
     adaptor->Run(op, opt_level, verify);
   } else {
     if (instrumentor) instrumentor->RunBeforePass(pass, op);
+    paddle::string::PrettyLogH1("--- Running analysis [%s]",
```
Comment (yuanlehome) [inline paddle/pir/pass/pass.cc]: --- Running analysis修改为--- Running PIR pass

**Review by zyfncg — DISMISSED** [review]

**Pull request closed without merging**